eth/api.go

```diff
@@ -715,6 +715,78 @@ func (api *ScrollAPI) GetNumSkippedTransactions(ctx context.Context) (uint64, er
 	return rawdb.ReadNumSkippedTransactions(api.eth.ChainDb()), nil
 }
 
+// FinalizedBlockHeight returns the L2 finalized block height.
+func (api *ScrollAPI) FinalizedBlockHeight(ctx context.Context) (uint64, error) {
+	finalizedBlockHeightPtr := rawdb.ReadFinalizedL2BlockNumber(api.eth.ChainDb())
+	if finalizedBlockHeightPtr == nil {
+		return 0, errors.New("L2 finalized block height not found in database")
+	}
+	return *finalizedBlockHeightPtr, nil
+}
+
+// IsBlockFinalized returns true if the provided block is finalized.
+func (api *ScrollAPI) IsBlockFinalized(ctx context.Context, blockHeight uint64) (bool, error) {
+	finalizedBlockHeightPtr := rawdb.ReadFinalizedL2BlockNumber(api.eth.ChainDb())
+	if finalizedBlockHeightPtr == nil {
+		return false, errors.New("L2 finalized block height not found in database")
+	}
+	return blockHeight <= *finalizedBlockHeightPtr, nil
```

Comment:
```suggestion
	finalizedBlockHeightPtr, err := api. FinalizedBlockHeight(ctx)
	if err != nil {
	    ..
	}
	return blockHeight <= *finalizedBlockHeightPtr, nil
```

Reply:
changed in [0b95a6a](https://github.com/scroll-tech/go-ethereum/pull/548/commits/0b95a6a75227b5ab6b469947cfafcb81afe4babf)